push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/util/JmsClient.java

```diff
@@ -207,8 +231,30 @@ public JmsSender inTransaction() {
 
         /**
          * Sets the property that can be later used to query message by selector.
+         *
+         * @param name of property
+         * @param value of property
+         * @return this sender object
          */
         public JmsSender withProperty(String name, String value) {
+            if (value == null) {
+                throw new NullPointerException("property value");
```

Comment:
The error message is a bit unclear, what do you think about `Parameter 'value' can't be null` or `Property value can't be null`. I'm aware it is already a NullPointerException and may look redundant but still, the message reads awkward.